Workbooks/Network Insights/ApplicationGatewayWebApplicationFirewallPolicies/AppGwWafDefault/AppGwWafDefault.workbook

```diff
@@ -677,364 +1274,1012 @@
-                      "linkIsContextBlade": true,
-                      "bladeOpenContext": {
-                        "bladeName": "LogsBlade",
-                        "extensionName": "Microsoft_Azure_Monitoring_Logs",
-                        "bladeParameters": [
-                          {
-                            "name": "resourceId",
-                            "source": "parameter",
-                            "value": "workspace"
+                        },
+                        "customWidth": "80",
+                        "name": "query - 113"
+                      },
+                      {
+                        "type": 3,
+                        "content": {
+                          "version": "KqlItem/1.0",
+                          "query": "let SelectedMS = dynamic({MessageFilter}); // reference to the above list of messages (Event trigger)\r\nlet Child = SelectedMS.Rule; // Used to choose a group of messages - redirects to the message which was grouped\r\nlet FakeData = (datatable (Message:string, ruleName_s:string, clientIp_s:string, clientIP_s:string, action_s:string, transactionId_g:string) [ \"\", \"\", \"\", \"\", \"\", \"\" ]);\r\nFakeData | union AzureDiagnostics\r\n| where OperationName == \"ApplicationGatewayFirewall\" \r\n| where '{RequestURI}' contains requestUri_s or '{RequestURI}' == ''\r\n| where '{SelectedRuleGroup}' contains ruleGroup_s or '{SelectedRuleGroup}' == ''\r\n| extend Rule = strcat(ruleName_s, Message), ClientIP = strcat(clientIp_s, clientIP_s)\r\n| extend Action = iif(action_s == \"Detected\", Action = \"Logged\", action_s)\r\n| extend TrackingID = strcat(transactionId_g)\r\n| where '{SelectedTrackingID}' == TrackingID or '{SelectedTrackingID}' == \"*\" \r\n| where Rule == Child or Child == \"*\"\r\n| where '{SelectedAction}' contains Action or '{SelectedAction}' == \"\"\r\n| where '{Selected}' == Rule or '{Selected}' == \"*\"\r\n| summarize count() by ClientIP\r\n| top 10 by count_ desc",
+                          "size": 0,
+                          "showAnalytics": true,
+                          "title": "Top 10 source IPs (select to filter)",
+                          "noDataMessage": "No data found –try to expand the detection time range to include more events",
+                          "timeContextFromParameter": "detectionTime",
+                          "exportFieldName": "x",
+                          "exportParameterName": "ClientIP",
+                          "exportDefaultValue": "*",
+                          "showExportToExcel": true,
+                          "exportToExcelOptions": "all",
+                          "queryType": 0,
+                          "resourceType": "microsoft.network/applicationgateways",
+                          "crossComponentResources": [
+                            "{Associations}"
+                          ],
+                          "visualization": "barchart",

```

Comment:
please remove extra visualization settings if not being used - eg. tile settings, map settings, graph settings